Xamarin.Forms.Platform.GTK/VisualElementTracker.cs

```diff
@@ -349,21 +350,47 @@ private void OnContainerButtonPressEvent(object o, ButtonPressEventArgs args)
 			if (view == null)
 				return;
 
-			if (args.Event.Type == Gdk.EventType.TwoButtonPress)
+			int numClicks = 0;
+			switch (args.Event.Type)
 			{
-				IEnumerable<TapGestureRecognizer> doubleTapGestures = view.GestureRecognizers
-					.GetGesturesFor<TapGestureRecognizer>(g => g.NumberOfTapsRequired == 2);
-
-				foreach (TapGestureRecognizer recognizer in doubleTapGestures)
-					recognizer.SendTapped(view);
+				case Gdk.EventType.ThreeButtonPress:
+					numClicks = 3;
+					break;
+				case Gdk.EventType.TwoButtonPress:
+					numClicks = 2;
+					break;
+				case Gdk.EventType.ButtonPress:
+					numClicks = 1;
+					break;
+				default:
+					return;
 			}
-			else
+
+			// Taps or Clicks
+			if (button == 1)
 			{
 				IEnumerable<TapGestureRecognizer> tapGestures = view.GestureRecognizers
-					.GetGesturesFor<TapGestureRecognizer>(g => g.NumberOfTapsRequired == 1);
+					.GetGesturesFor<TapGestureRecognizer>(g => g.NumberOfTapsRequired == numClicks);
 
 				foreach (TapGestureRecognizer recognizer in tapGestures)
 					recognizer.SendTapped(view);
+
+				IEnumerable<ClickGestureRecognizer> clickGestures = view.GestureRecognizers
+					.GetGesturesFor<ClickGestureRecognizer>(g => g.NumberOfClicksRequired == numClicks &&
+															g.Buttons == ButtonsMask.Primary);
+
+				foreach (ClickGestureRecognizer recognizer in clickGestures)
+					recognizer.SendClicked(view, ButtonsMask.Primary);
+			}
+			else
+			{
+				// right click
+				IEnumerable<ClickGestureRecognizer> rigthClickGestures = view.GestureRecognizers
```

Comment:
rigthClickGestures >> rightClickGestures

Reply:
changed and rebased on top of master